src/views/tutorial-view/server.ts

```diff
@@ -34,14 +37,33 @@ export async function getTutorialPageProps(
    * from the params. So we can assume `slug` index 1 is always the tutorial name
    * */
   const [collectionFilename, tutorialFilename] = slug
-  const tutorialDbSlug = `${product.slug}/${tutorialFilename}`
-  const baseTutorialData = await getTutorial(tutorialDbSlug)
+  const collectionDbSlug = `${product.slug}/${collectionFilename}`
+  const currentCollection = await getCollection(collectionDbSlug)
+  /**
+   * We need to get the database slug for this tutorial, which may belong in a different
+   * product directory in the filesystem. For example a tutorial with slug : `consul/get-started`
+   * may be reference in a vault collection. So we can't assume this current
+   * product context is valid for the db slug path
+   *
+   * @TODO - FOR PROD We will add a check in the content sync that
+   * ensures no two files in a collection have the same filename
+   */
+  const currentTutorial = currentCollection.tutorials.find((t) =>
```

Comment:
Another fix here related to the cross-product tutorial issue. In order to query for the tutorial data, we need either the proper database slug (the tutorials filesystem path) or the tutorial id. In order to get the tutorial slug, we need to call for the collection and find that tutorial based on the filename. 

We're planning on adding a check in the content sync that prevents files with the exact same name being listed in the same collection (currently not a problem).